### DISCUSSION TITLE
Support for Apple Home Key?

### DISCUSSION THREAD

**Discussion by guw** [leosac/leosac#144] [Q&A]
Is it possible to create a solution supporting Apple Home Key with leosac?

**Comment by Maxhy** [comment]
In theory yes as it is NFC based and probably built on top of ISO7816. But as usual with Apple there is restrictions...
With their current model, a proper partnership with Apple from Leosac company would be required before performing the technical implementation. Open source model may be a blocker for Apple here, I'm not sure about the details at this time.
Honestly I would recommend to avoid such closed ecosystem credential. You may only regret it in the long run.

**Reply by guw** [reply]
I prefer the UX and easy of convenience. Hence, avoidance is not really an option to me. :) 

Anyway, I made some progress with my research and I believe it may work out of the box. You need a reader certified for Apple VAS (or Google SmartTap if you like Android). For example, the VTAP100 from Dot Origin is such a device and has a Wiegand interface. I think this could work. Of course, you need to sign up with Apple Developer ID and create a key and certificate for issuing and validating passes but there are services (eg., [passkit](https://help.passkit.com/en/articles/4430727-enabling-and-using-nfc-passes)), which simplify this. I have not found an OSS app yet but it's probably doable.

**Reply by Maxhy** [reply]
Ah yes if the communication doesn't needs to be computed by Leosac (eg. like in "transparent communication") then it should be straightforward as you pointed out.
